src/mmrelay/message_queue.py

```
@@ -238,15 +242,58 @@ def enqueue(
+                    # Success - break out of loop
+                    break
+                except Full:
+                    if wait:
+                        # Check if we've exceeded timeout
+                        if timeout is not None:
+                            elapsed = time.monotonic() - start_time
+                            if elapsed >= timeout:
+                                logger.error(
+                                    f"Message queue full ({self._queue.qsize()}/{MAX_QUEUE_SIZE}) "
+                                    f"and wait timed out after {timeout}s, dropping message: {description}"
+                                )
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)